flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/observer/Observer.java

```diff
@@ -129,13 +135,55 @@ private boolean observeFlinkJobStatus(FlinkDeployment flinkApp, Configuration ef
         }
     }
 
+    public boolean observeSavepointStatus(FlinkDeployment flinkApp, Configuration effectiveConfig) {
+        SavepointInfo savepointInfo = flinkApp.getStatus().getJobStatus().getSavepointInfo();
+        if (savepointInfo.getTriggerId() == null) {
+            LOG.info("Checkpointing not in progress");
+            return true;
+        }
+        SavepointFetchResult savepointFetchResult;
+        try {
+            savepointFetchResult = flinkService.fetchSavepointInfo(flinkApp, effectiveConfig);
+        } catch (Exception e) {
+            LOG.error("Exception while fetching savepoint info", e);
+            return false;
+        }
+
+        if (!savepointFetchResult.isTriggered()) {
+            String error = savepointFetchResult.getError();
+            if (error != null
+                    || !SavepointUtils.withinGracePeriod(operatorConfiguration, savepointInfo)) {
+                LOG.info("Savepoint not triggered, setting reconciliation error");
+                savepointInfo.setTriggerId(null);
+                ReconciliationStatus reconciliationStatus =
+                        flinkApp.getStatus().getReconciliationStatus();
+                reconciliationStatus.setSuccess(false);
+                reconciliationStatus.setError(error != null ? error : "Savepoint status unknown");
```

Comment:
We should move the FlinkDeploymentController#updateForReconciliationError into a static utility and reuse here

Reply:
done